src/components/Approval/Approval.tsx

```diff
@@ -157,6 +146,38 @@ export const Approval = () => {
     showErrorToast,
   ])
 
+  const toggleIsExactAllowance = useCallback(() => {
+    setIsExactAllowance(!isExactAllowance)
+  }, [isExactAllowance])
+
+  useEffect(() => {
+    setBuildApprovalTxInput(undefined)
+    createBuildApprovalTxInput(isExactAllowance)
+      .then(buildApprovalTxInput => {
+        setBuildApprovalTxInput(buildApprovalTxInput)
+      })
+      .catch(e => {
+        showErrorToast(e)
+        return history.push(TradeRoutePaths.Input)
+      })
+  }, [isExactAllowance, createBuildApprovalTxInput, history, showErrorToast])
```

Comment:
do we have a better pattern for this i should be using? like rtkquery or thunk? we'd later probably want to trigger this on an interval so gas estimate is always <60s old

Reply:
We have a custom `useInterval` hook though I can see no current usages of it, not sure how reliable it is - with that being said, I'd probably call this an edge case, not sure if it's worth the hassle.

Reply:
even better will be gas oracle websocket at some point to trigger push updates on gas estimates :) agree would just keep it as it stands for now especially with this big of a refactor